webpack.config.js

```diff
@@ -488,14 +488,28 @@ async function initializeModuleFederation() {
             // Other containers will be forced to use the exact versions of shared modules that the web app provides.
             makeSingletonSharedModules([
                 'react',
+                'react-beautiful-dnd',
                 'react-bootstrap',
                 'react-dom',
                 'react-intl',
                 'react-redux',
                 'react-router-dom',
             ]),
         ],
-    }));
+    };
+
+    // Desktop specific code for remote module loading
+    moduleFederationPluginOptions.exposes = {
+        './root': 'components/root',
+        './crtWatcher': 'components/threading/channel_threads/posts_channel_reset_watcher',
+        './store': 'stores/redux_store.jsx',
+        './styles': 'sass/styles.scss',
```

Comment:
Do these 4 actually need to be exposed at the moment? For the Redux store in particular, I'd still hope we could avoid accessing it outside of a React context, and if we need it anywhere else, we could even pass it in separately.

Reply:
Yes, all 4 need to be exposed:
- `root` and `crtWatcher` need to be rendered in order to run the app
- `styles` needs to be imported to make sure the styles aren't overridden by desktop styles. This could change in the future
- `store` needs to be sent to the Redux provider which is part of my rendering code

Reply:
- For `root` and `crtWatcher`, I think it would make more sense to have the desktop app use the `App` component from the web app for better abstraction or even going a step further and exposing something like the web app's `entry.tsx`. In particular, needing to expose the `crtWatcher` feels like we're leaking some details that the desktop app shouldn't need to care about.
- For `styles`, could you elaborate a bit more about the problem? I'm not sure how exposing the styles object would change whether or not the desktop app is overwriting web app styles. It also only includes the globally applied styles and not anything controlled by `styled-components` or (I think) by CSS imported directly into the corresponding component.
- For `store`, I would've thought that everything that relies on web app state would be rendered inside of the web app and therefore covered by its `Provider`. If there's anything rendered outside of that scope, I would assume it would need its own state tree anyway